apps/fxlab/app/src/main/java/com/mobileer/androidfxlab/MainActivity.kt

```diff
@@ -98,22 +98,32 @@ class MainActivity : AppCompatActivity() {
         if (Build.VERSION.SDK_INT > Build.VERSION_CODES.M) {
             handleMidiDevices()
         }
+
+    }
+
+    override fun onDestroy() {
+        // Clear the FX UI
+        EffectsAdapter.effectList.clear()
+        EffectsAdapter.notifyDataSetChanged()
+        super.onDestroy()
+    }
+
+    override fun onPause() {
+        // Shutdown Engine
+        NativeInterface.destroyAudioEngine()
+
+        super.onPause()
```

Comment:
This order is correct. super.onPause() at the end.

Reply:
Done.